packages/twenty-server/src/engine/metadata-modules/remote-server/remote-server.service.ts

```diff
@@ -117,23 +117,32 @@ export class RemoteServerService<T extends RemoteServerType> {
       throw new NotFoundException('Object does not exist');
     }
 
-    const remoteTablesToRemove = (
-      await this.remoteTableService.findAvailableRemoteTablesByServerId(
-        id,
-        workspaceId,
-      )
-    ).filter((remoteTable) => remoteTable.status === RemoteTableStatus.SYNCED);
-
-    if (remoteTablesToRemove.length) {
-      for (const remoteTable of remoteTablesToRemove) {
-        await this.remoteTableService.unsyncRemoteTable(
-          {
-            remoteServerId: id,
-            name: remoteTable.name,
-          },
+    try {
+      const remoteTablesToRemove = (
+        await this.remoteTableService.findAvailableRemoteTablesByServerId(
+          id,
           workspaceId,
-        );
+        )
+      ).filter(
+        (remoteTable) => remoteTable.status === RemoteTableStatus.SYNCED,
+      );
+
+      if (remoteTablesToRemove.length) {
+        for (const remoteTable of remoteTablesToRemove) {
+          await this.remoteTableService.unsyncRemoteTable(
+            {
+              remoteServerId: id,
+              name: remoteTable.name,
+            },
+            workspaceId,
+          );
+        }
       }
+    } catch (error) {
+      console.error(
```

Comment:
this could be quite problematic because we will keep objectMetadata that point on nothing. We need to handle this case either in the unSyncRemoteTable function or in the catch

Reply:
We do not store the server id on metadata today. The only thing that can help retrieving metadata is the name. If we cannot access the remote database, we cannot find the remote table names. Only solution I see is either storing remote table objects, either storing the info on the metadata

Reply:
Forget it. I now fetch the foreign tables instead of calling distant schema. This allows me to filter based on server id.